hazelcast/src/main/java/com/hazelcast/cache/impl/nearcache/impl/DefaultNearCache.java

```diff
@@ -128,6 +140,9 @@ public void destroy() {
         if (expirationTaskFuture != null) {
             expirationTaskFuture.cancel(true);
         }
+        if (storageTaskFuture != null) {
+            storageTaskFuture.cancel(true);
```

Comment:
Yes, I was thinking about this. Have to check how to implement this properly. I hope the renaming solution will be atomic enough until then.